src/flags.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+// -------------------------------------------------------------------------------
+// No requires or imports since this is loaded early in the cli lifecycle and we
+// want to minimize the number of packages that load before enabling require
+// instrumentation.
+// -------------------------------------------------------------------------------
+
+export interface ProcessLike {
+  argv: string[];
+  env: { [key: string]: string | undefined };
+}
+
+export function preprocessCliFlags(process: ProcessLike): void {
+  process.argv = process.argv.filter((arg, index) => {
+    switch (arg) {
+      case '--dev-debug': {
+        let debug = '*';
+        if (process.argv[index + 1] && !process.argv[index + 1]?.startsWith('-')) {
+          // if the next arg is defined and IS NOT a flag, set it as the logger level
+          debug = process.argv[index + 1];
+        }
+
+        // convert --dev-debug into a set of environment variables
+        process.env.DEBUG = debug;
+        process.env.SF_DEBUG = '1';
+        process.env.SF_ENV = 'development';
```

Comment:
are these env var documented?

I don't see any ref to `SF_DEBUG`:
https://github.com/search?q=SF_DEBUG+org%3Asalesforcecli+&type=code

for `SF_ENV` only this:
https://github.com/search?q=SF_ENV%20org%3Asalesforcecli%20&type=code

Reply:
I converted them from `SFDX_` to `SF_`, but didn't think to use that global GH search. 

I don't see useful references for `SFDX_DEBUG`, 

There's a few more hits for `SFDX_ENV` : https://github.com/search?q=SFDX_ENV+org%3Asalesforcecli+&type=code

but I'm not sure we need either of them 🤔